nexus/src/db/datastore/ip_pool.rs

```diff
@@ -158,17 +159,45 @@ impl DataStore {
         };
         let pool = IpPool::new(&new_pool.identity, project_id, rack_id);
         let pool_name = pool.name().as_str().to_string();
-        diesel::insert_into(dsl::ip_pool)
-            .values(pool)
-            .returning(IpPool::as_returning())
-            .get_result_async(self.pool_authorized(opctx).await?)
+
+        if let Some(project_id) = project_id {
+            Project::insert_resource(
```

Comment:
This is probably not a new issue, but do we not need to do an authz check here?

If we did, then at L171 I think you could use `authz_resource.not_found()`.